### PR TITLE
Fix wrong rates data after switching store view (cart page)

### LINKED ISSUE #31444: [Issue] Fix wrong rates data after switching store view (cart page)
This issue is automatically created based on existing pull request: magento/magento2#31354: Fix wrong rates data after switching store view (cart page)

---------
### Description (*)

In cart page, after switching store view, shipping title in summary block is incorrect

Reproduced from 2.3.x

### Fixed Issues (if relevant)
1. Fixes magento/magento2#<issue_number>

### Manual testing scenarios (*)

1. [BO] Create another store view (Example: modify the default store view to set the code to "en", then create a new store view with the code "fr")

2. [BO] Enable store view code in URL

3. ~[BO] Keep one delivery method (example: keep flat rate)~
   _After verification is not necessary to keep one delivery method_

4. [BO] In French store, change delivery method title and clear cache

5. In front (default store view), add product in cart

6. Go to the cart page and change store view

7. Title of delivery method does not contain title set for FR store

#### Before fix

**Default store (english)**

Screenshot_2020-12-19 Shopping Cart

**Switching to French store**

Screenshot_2020-12-19 Shopping Cart(1)

#### With fix

**Switching to French store**

Screenshot_2020-12-19 Shopping Cart(2)

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by Coosos**
### Description (*)

In cart page, after switching store view, shipping title in summary block is incorrect

Reproduced from 2.3.x

### Fixed Issues (if relevant)
1. Fixes magento/magento2#<issue_number>

### Manual testing scenarios (*)

1. [BO] Create another store view (Example: modify the default store view to set the code to "en", then create a new store view with the code "fr")

2. [BO] Enable store view code in URL

3. ~[BO] Keep one delivery method (example: keep flat rate)~
   _After verification is not necessary to keep one delivery method_

4. [BO] In French store, change delivery method title and clear cache

5. In front (default store view), add product in cart

6. Go to the cart page and change store view

7. Title of delivery method does not contain title set for FR store

#### Before fix

**Default store (english)**

Screenshot_2020-12-19 Shopping Cart

**Switching to French store**

Screenshot_2020-12-19 Shopping Cart(1)

#### With fix

**Switching to French store**

Screenshot_2020-12-19 Shopping Cart(2)

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#31444: Fix wrong rates data after switching store view (cart page)

### CODE REVIEW

**Comment by sidolov** [comment]
@magento create issue

**Review by gabrieldagama — CHANGES_REQUESTED** [review]
Hi @Coosos, thanks for contributing! Can you cover these changes with automated tests? 

Thanks!

**Comment by engcom-Hotel** [comment]
Hello @Coosos,
I'll help with the test.

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@Coosos,
Could you please submit  'Magento Contributor Agreement'
https://devdocs.magento.com/contributor-guide/contributing.html#contribute

**Comment by Coosos** [comment]
Hello 

@engcom-Hotel : I signed Adobe CLA :smile: 

@gabrieldagama Having a little help for the tests would not be refused because I have never done any on magento :disappointed:

I know that this file needs to be updated `dev/tests/js/jasmine/tests/app/code/Magento/Checkout/frontend/js/model/cart/totals-processor/default.test.js` to add `storeCode: 1` at line 35 / 36

But I don't know how I can do a test to check this case.

**Comment by gabrieldagama** [comment]
@magento run all tests

**Comment by gabrieldagama** [comment]
Hi @Coosos, sure thing!

Probably MFTF test would be more suitable in this case, are you interested in learning that? More details can be found here: https://devdocs.magento.com/mftf/docs/introduction.html

Please let me know if you would like to do that, otherwise, we can try to add test coverage to your PR.

Thanks!

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by Coosos** [comment]
Thank you @engcom-Hotel for the test.
On the other hand, shouldn't the "dev / tests / js / jasmine / tests / app / code / Magento / Checkout / frontend / js / model / cart / totals-processor / default.test.js" file be modified?

@gabrieldagama Sorry for yesterday, I didn't have time to make a return. 
I was going to ask you to do it because I would have taken too long to learn how to test.

**Comment by engcom-Hotel** [comment]
HI @Coosos,
Glad to help.
Probably, please check unit tests and fix it.
Let me know if you need help with it. 
Thank you.

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by Coosos** [comment]
@engcom-Hotel How download unit report ? 
_I will see after work to launch the phpunit locally_

**Comment by engcom-Hotel** [comment]
@Coosos ,
You can open tests details in the tests list, select correspondent test and open Build reports.

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by Coosos** [comment]
@engcom-Hotel If it is the "details" button on "Unit Tests" line which redirects to "http://10.234.200.109:8080/job/Unit-Tests/9877", the page is not accessible.

I will run the unit tests locally

**Comment by Coosos** [comment]
@engcom-Hotel 
I don't understand, but when I run the command "vendor/bin/phpunit -c dev/tests/unit/phpunit.xml.dist", I do have some failed tests but phpunit shows me no list. Same with "bin/magento dev:tests:run unit" command, I have several in failure but it only tells me the command in failure:

```
WARNINGS!
Tests: 238, Assertions: 440, Warnings: 16, Skipped: 1.
----------------------------------------------------------------------
FAILED - 1 of 3:
 - /var/www/magento2/dev/tests/unit> /usr/local/bin/php /var/www/magento2/./vendor/phpunit/phpunit/phpunit
```

On the other hand, I had to apply this MR (https://github.com/magento/magento2/pull/31296) otherwise I had errors (Use of undefined constant GLOB_BRACE)

**Comment by engcom-Hotel** [comment]
@Coosos I'll check it.

**Comment by Coosos** [comment]
@magento run all tests

**Comment by Coosos** [comment]
@magento run Functional Tests EE

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests EE

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by Coosos** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Functional Tests EE

**Comment by Coosos** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by Coosos** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by Coosos** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by engcom-Hotel** [comment]
Looks like MFTF failed tests are not related to changes.

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by Coosos** [comment]
@magento run WebAPI Tests, Functional Tests EE, Functional Tests CE, Functional Tests B2B

**Comment by Coosos** [comment]
@magento run Functional Tests B2B

**Comment by Coosos** [comment]
@magento run Functional Tests B2B

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B

**Comment by Coosos** [comment]
@magento run Functional Tests B2B

**Comment by Coosos** [comment]
Complicated to validate the tests, and the errors are never the same

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B

**Comment by Coosos** [comment]
@magento run Functional Tests B2B

**Comment by Coosos** [comment]
@magento run Functional Tests B2B

**Comment by Coosos** [comment]
@magento run Functional Tests B2B

**Comment by Coosos** [comment]
@magento run Functional Tests B2B

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B

**Comment by engcom-Hotel** [comment]
@magento run Database Compare, Functional Tests CE, Functional Tests EE, Integration Tests, Magento Health Index, Sample Data Tests B2B, Sample Data Tests CE, Sample Data Tests EE, Semantic Version Checker, Static Tests, Unit Tests, WebAPI Tests

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests CE

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests CE

File: app/code/Magento/Sales/Test/Mftf/Test/StorefrontVerifyStoreViewCartShippingNameTest.xml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            <actionGroup ref="AdminDeleteStoreViewActionGroup" stepKey="deleteStoreView"/>
+            <actionGroup ref="AdminLogoutActionGroup" stepKey="logout"/>
+            <magentoCLI command="config:set {{StorefrontDisableAddStoreCodeToUrls.path}} {{StorefrontDisableAddStoreCodeToUrls.value}}" stepKey="disableAddStoreCodeToUrls"/>
+        </after>
+        <actionGroup ref="AdminOpenShippingMethodsConfigPageActionGroup" stepKey="openShippingMethodConfigPage"/>
+        <actionGroup ref="AdminSwitchStoreViewActionGroup" stepKey="switchStoreViewAdmin"/>
+        <conditionalClick selector="{{AdminShippingMethodFlatRateSection.carriersFlatRateTab}}" dependentSelector="{{AdminShippingMethodFlatRateSection.carriersFlatRateTitle}}" visible="false" stepKey="expandTab"/>
+        <click selector="{{AdminShippingMethodFlatRateSection.carriersFlatRateTitle}}" stepKey="uncheckUseSystemValueForFlatRateTitle"/>
+        <fillField selector="{{AdminShippingMethodFlatRateSection.carriersFlatRateTitleField}}" userInput="Test carrier title name" stepKey="fillFieldTitle"/>
+        <click selector="{{AdminShippingMethodFlatRateSection.carriersSaveConfig}}" stepKey="saveShippingConfig"/>
```
Comment (gabrieldagama) [inline app/code/Magento/Sales/Test/Mftf/Test/StorefrontVerifyStoreViewCartShippingNameTest.xml]: Please create an action group on the Shipping module for this, make sure to have the carrier title name as parameter.

File: app/code/Magento/Sales/Test/Mftf/Test/StorefrontVerifyStoreViewCartShippingNameTest.xml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        <actionGroup ref="CliCacheFlushActionGroup" stepKey="cleanCache">
+            <argument name="tags" value=""/>
+        </actionGroup>
```
Comment (gabrieldagama) [inline app/code/Magento/Sales/Test/Mftf/Test/StorefrontVerifyStoreViewCartShippingNameTest.xml]: Have you tested this without flushing cache? If that is absolutely required, please just clean config cache, not the whole cache.

File: app/code/Magento/Sales/Test/Mftf/Test/StorefrontVerifyStoreViewCartShippingNameTest.xml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            <argument name="product" value="$$createSimpleProduct$$"/>
+        </actionGroup>
+        <actionGroup ref="StorefrontProductPageAddSimpleProductToCartActionGroup" stepKey="addProductToCart"/>
+        <amOnPage url="{{CheckoutCartPage.url}}" stepKey="openShoppingCart"/>
+        <actionGroup ref="StorefrontSwitchStoreViewActionGroup" stepKey="cartSwitchStoreView"/>
+        <grabTextFrom selector="{{CheckoutPaymentSection.orderSummaryShippingMethod}}" stepKey="getShippingName"/>
+        <assertStringContainsString stepKey="assertShippingTitle">
+            <expectedResult type="string">Test carrier title name</expectedResult>
+            <actualResult type="variable">getShippingName</actualResult>
+        </assertStringContainsString>
```
Comment (gabrieldagama) [inline app/code/Magento/Sales/Test/Mftf/Test/StorefrontVerifyStoreViewCartShippingNameTest.xml]: Create a action group for this one as well on shipping module.

**Review by gabrieldagama — CHANGES_REQUESTED** [review]
Hi @engcom-Hotel, thanks for working on it, please have a look on my comments below.

File: app/code/Magento/Sales/Test/Mftf/Test/StorefrontVerifyStoreViewCartShippingNameTest.xml (hunk 1 above)
Comment (engcom-Hotel) [reply app/code/Magento/Sales/Test/Mftf/Test/StorefrontVerifyStoreViewCartShippingNameTest.xml]: Has been created.

File: app/code/Magento/Sales/Test/Mftf/Test/StorefrontVerifyStoreViewCartShippingNameTest.xml (hunk 2 above)
Comment (engcom-Hotel) [reply app/code/Magento/Sales/Test/Mftf/Test/StorefrontVerifyStoreViewCartShippingNameTest.xml]: Has been removed.

File: app/code/Magento/Sales/Test/Mftf/Test/StorefrontVerifyStoreViewCartShippingNameTest.xml (hunk 3 above)
Comment (engcom-Hotel) [reply app/code/Magento/Sales/Test/Mftf/Test/StorefrontVerifyStoreViewCartShippingNameTest.xml]: Has been created.

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests EE

**Comment by Coosos** [comment]
@magento run Functional Tests EE

**Review by gabrieldagama — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @gabrieldagama, thank you for the review.
[ENGCOM-8836](https://jira.corp.magento.com/browse/ENGCOM-8836) has been created to process this Pull Request
:eight_spoked_asterisk: @gabrieldagama, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by magento-engcom-team** [comment]
@Coosos thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by engcom-Bravo** [comment]
@gabrieldagama The original issue is not reproducible on the latest 2.4-develop. After changing the Store View, the Shipping method name is also changed.
fr_not_repro

Is there a need to Merge this PR?
Thank you in advance

**Comment by engcom-Bravo** [comment]
@Coosos Thank you for your contribution. The original issue is not reproducible on the latest 2.4-develop, as you can see from the comment above. It has probably been fixed in scope of another delivery.
Unfortunately, we have to close this PR as we can not Merge a PR in case when the original issue is not reproducible.
Thank you for your collaboration.

**Comment by Coosos** [comment]
I will check, because I do not see which MR to be able to fix this (knowing that I tested 2.4.2)

**Pull request closed without merging**